kawin/Thermodynamics.py

```diff
@@ -1410,20 +1560,18 @@ def curvatureFactor(self, x, T, precPhase = None, training = False):
         '''
         if precPhase is None:
             precPhase = self.phases[1]
-        if hasattr(x, '__len__'):
-            #Remove first element if x lists composition of all elements
-            if len(x) == len(self.elements) - 1:
-                x = x[1:]
+        if not hasattr(x, '__len__'):
+            x = [x]
 
-            conds = {v.X(self.elements[i+1]): x[i] for i in range(len(x))}
-        else:
-            conds = {v.X(self.elements[1]): x}
-        conds.update({v.N: 1, v.P: 1e5, v.GE: 0, v.T: T})
+        #Remove first element if x lists composition of all elements
+        if len(x) == len(self.elements) - 1:
+            x = x[1:]
+        cond = self._getConditions(x, T, 0)
 
         #Perform equilibrium from scratch if cache not set or when training surrogate
         if self._compset_cache.get(precPhase, None) is None or training:
             eq = self.getEq(x, T, 0, precPhase)
-            state_variables = np.array([conds[v.GE], conds[v.N], conds[v.P], conds[v.T]], dtype=np.float64)
+            state_variables = np.array([cond[v.GE], cond[v.N], cond[v.P], cond[v.T]], dtype=np.float64)
```

Comment:
If you want to avoid hard-coding assumptions of N-P-T-GE, you can use `compset.phase_record.state_variables` to get a list of `StateVariable` objects. Then you could do
```python
state_variables = np.array([cond[sv] for sv in compset.phase_record.state_variables], dtype=np.float64)
```